### DISCUSSION TITLE
Quire PDF - failed to connect to localhost: Connection refused

### DISCUSSION THREAD

**Discussion by salger1** [thegetty/quire#272] [Get Help] [answered]
I'm trying to run quire PDF for our project but am getting an error output that says "error: Failed to connect to localhost port 1313: Connection refused". I attached the verbose output. 

Things I tried: 
- re-downloading Prince (version 14.2)
- updating my mac (Monterey 12.3)
- checked firewall settings (firewall was off when I ran the attached command)
- checked for files with a space in the title (per https://github.com/thegetty/quire/discussions/208)
- checked that the mac was allowing apps downloaded from identified developers
- removed line 10 from config.yml: baseURL: "https://nelson-atkins.org/fpc" (per email from Greg in December 2020 about a different PDF issue)

Things I have not yet tried: 
- Working backwards from the last known functioning build. The last pdf build I attempted was in October so this will be quite a process. Hoping someone will know what is wrong without going that far back. 

[quire pdf --verbose.txt](https://github.com/thegetty/quire/files/8327796/quire.pdf.--verbose.txt)

**Comment by geealbers** [comment]
hi @salger1. Do you know if you've updated Quire since that working October build? And when you get this error, if you go to http://localhost:1313/ in your browser do you see anything?

**Comment by salger1** [comment, marked as answer]
Excellent question @geealbers I just looked through our project history and I don't think we have updated Quire since the October build. I believe our last update was in July of last year when we started working on incorporating IIIF.

**Reply by Erin-Cecele** [reply]
Hi @salger1 - are you still experiencing issues?

**Reply by salger1** [reply]
Hi @Erin-Cecele Unfortunately yes, I am still having the pdf build issue.

**Reply by geealbers** [reply]
@salger1, I may have a fix:

1. In your Finder, open the "Go to folder" window with Shift-Command-G

2. Copy in this string and hit enter: /usr/local/lib/node_modules/@thegetty/quire-cli/bin

3. Move the index.js file there onto your desktop in case you need to revert back to it, and then download, unzip, and save this attached version of index.js in its place: [**index.js.zip**](https://github.com/thegetty/quire/files/8557661/index.js.zip)

4. Run `quire pdf`

If it still doesn't work, revert your index.js file back to the original one, and then run `quire pdf --verbose` and share the full output with us here


.

**Reply by geealbers** [reply]
@salger1, I realized a probable flaw in my fix above, in that you may need to reset the permissions on the new index.js file for it to run. To do that, in your Terminal, run: chmod 755 /usr/local/lib/node_modules/@thegetty/quire-cli/bin/index.js

**Reply by salger1** [reply]
@geealbers It worked! Thank you for the help.
